### DISCUSSION TITLE
When mounting a ref in a table, what type of function should be used?

### DISCUSSION THREAD

**Discussion by 389653475** [element-plus/element-plus#18291] [Q&A]
I need to display multiple tables dynamically, and I also need to control the expansion and collapse of the tables, so I dynamically hang ref, the code is as follows

`<el-table
          :ref="(el) => setItemRef(el, InfoIndex)"
          v-loading="loading"
          :data="InfoListItem.children"
          style="width: 100%; margin-bottom: 20px"
          row-key="category"
          border
          :tree-props="{ hasChildren: 'hasChildren', children: 'children', checkStrictly: false }"
          :default-sort="{ prop: 'totalConsumed', order: 'descending' }"
        >
const setItemRef = (el: any, index: number) => {
  if (el) {
    tableRefs.value[`ElTableRef_${index}`] = el
  }
}
const tableRefs = ref<{ [key: string]: InstanceType<typeof ElTable> }>({})`
I would like to know what the type of el should be in the setItemRef function?

**Comment by warmthsea** [comment]
```ts
el: globalThis.ComponentPublicInstance<InstanceType<typeof ElTable>>
```
